src/daft-core/src/series/mod.rs

```diff
@@ -232,3 +232,48 @@ impl Series {
         }
     }
 }
+
+pub trait NamedFrom<T, Phantom: ?Sized> {
+    /// Initialize by name and values.
+    fn new(name: &str, _: T) -> Self;
```

Comment:
could be clearer if this was `fn new(name: &str, _: Vec<T>) -> Self;`

Reply:
so the idea was that we could impl it for anything, not just `Vec<T>`. Right now it is only implemented for `Vec<T>` because that's what I needed at the moment.